src/main/java/org/folio/qm/converter/MarcDtoConverter.java

```diff
@@ -52,10 +52,11 @@ public QuickMarcView convert(@NonNull ParsedRecordDto source) {
     var format = typeMapper.fromDto(source.getRecordType());
     var leader = convertLeader(marcRecord);
     var fields = fieldsConverter.convertDtoFields(marcRecord.getVariableFields(), marcRecord.getLeader(), format);
+    var reorderedFields = fieldsConverter.reorderFieldsBasedOnParsedRecordOrder(fields, parsedRecord);
```

Comment:
Can't we just keep order during "convertDtoFields" on a previous line?
Looks like this method takes control fields first and then data fields. I suppose we can preserve order by having one stream with switch/case in "map" method instead of two streams inside "convertDtoFields"

Reply:
No, because the extractMarcRecord(parsedRecord) method uses MarcJsonReader, which orders fields according to the MARC standard (Controlled fields first, then Data Fields). Therefore, marcRecord.getVariableFields() already is ordered.